util/Pending.h

```diff
@@ -127,6 +127,52 @@ namespace Pending {
         return Pending<decltype(parser(path))>(
             std::async(std::launch::async, parser, path), path.filename().string());
     }
+
+    /** Return a Pending<T> constructed with \p parser, \p arg1, and \p path*/
+    template <typename Func, typename Arg1>
+    auto Parsed(const Func& parser, const Arg1& arg1, const boost::filesystem::path& path)
+        -> Pending<decltype(parser(arg1, path))>
+    {
+        auto f = std::async(std::launch::deferred, parser, arg1, path);
+        f.wait();
```

Comment:
> auto f = std::async(std::launch::deferred, parser, arg1, path);

The deferred future is created

> f.wait();

Deferred future starts executing. 
Can we just use std::launch::async policy?

Also, `wait` is blocking, so this thread will wait until the future is finished.  Could we exit this method without waiting until the task finished?

Reply:
No, because of python we cannot use threads other than python interpreter was initialized.

Reply:
> No, because of python we cannot use threads other than python interpreter was initialized.

Is it about using async instead of deffered or about blocking call?

Also, the name Pending is a little bit confusing.  At this point, we are not pending anything we already called the wait method.

Reply:
Yes, it just about to use the same code to work with parsing result as before.